test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
+      # redirected to samples index
       assert_selector 'h1', text: I18n.t(:'projects.samples.index.title'), count: 1
+      # deleted sample row no longer exists
+      assert_no_selector "#samples-table table tbody tr[id='#{@sample1.id}']"
+      # remaining samples still appear on table
       assert_selector '#samples-table table tbody tr', count: 2
-      within('tbody tr:first-child th') do
-        assert_text @sample2.puid
-      end
+      ### VERIFY END ###
     end
 
-    test 'should destroy Sample from sample listing page' do
+    test 'should destroy Sample from samples table' do
+      ### SETUP START ###
       visit namespace_project_samples_url(@namespace, @project)
 
       assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 3, count: 3,
                                                                            locale: @user.locale))
-      table_row = find(:table_row, [@sample1.name])
+      ### SETUP END ###
 
-      within table_row do
-        click_link 'Remove'
+      ### ACTIONS START ###
+      within("tr[id='#{@sample1.id}']") do
+        # click remove action
+        click_link I18n.t('projects.samples.index.remove_button')
       end
 
-      within('dialog') do
+      # confirm destroy action
+      within('#dialog') do
         assert_text I18n.t('projects.samples.deletions.new_deletion_dialog.description', sample_name: @sample1.name)
         click_button I18n.t('projects.samples.deletions.new_deletion_dialog.submit_button')
       end
+      ### ACTIONS END ###
 
+      ### VERIFY START ###
+      # success flash msg
       assert_text I18n.t('projects.samples.deletions.destroy.success', sample_name: @sample1.name,
                                                                        project_name: @project.namespace.human_name)
-
-      assert_no_selector '#samples-table table tbody tr', text: @sample1.puid
-      assert_no_selector '#samples-table table tbody tr', text: @sample1.name
+      # sample no longer exists
+      assert_no_selector "tr[id='#{@sample1.id}']"
+      # still on samples index page
       assert_selector 'h1', text: I18n.t(:'projects.samples.index.title'), count: 1
-      assert_selector '#samples-table table tbody tr', count: 2
-      within('#samples-table table tbody tr:first-child th') do
-        assert_text @sample2.puid
-      end
+      # remaining samples still in table
+      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 2, count: 2,
+                                                                           locale: @user.locale))
+      ### VERIFY END ###
     end
 
-    test 'should transfer multiple samples' do
-      project2 = projects(:project2)
+    test 'transfer dialog sample listing' do
+      ### SETUP START ###
+      samples = @project.samples.pluck(:puid, :name)
       visit namespace_project_samples_url(@namespace, @project)
```

Comment:
**nitpick (blocking):** Can you add an assertion that the limit component is visible to ensure that the samples table has fully loaded to reduce flakes.

Reply:
Added in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac